pkg/orchestration/wiring/kubeingress/plugin.go

```diff
@@ -124,11 +125,9 @@ func buildServiceResource(deploymentName smith_v1.ResourceName, selectorLabels m
 	return serviceResource
 }
 
-// buildIngressResource constructs the Kube / KITT Ingress object
-// with a default rule, plus all aliases rules from dependants internalDNS resources
-func buildIngressResource(serviceName smith_v1.ResourceName, resource *orch_v1.StateResource, context *wiringplugin.WiringContext) (wiringplugin.WiredSmithResource, error) {
+func buildIngressResourceFromSpec(serviceName smith_v1.ResourceName, resourceName voyager.ResourceName, timeout int, context *wiringplugin.WiringContext) (*wiringplugin.WiredSmithResource, error) {
 	var ingressRules []ext_v1b1.IngressRule
-	ingressName := string(resource.Name) + "-" + ingressPostfix
+	ingressName := string(resourceName) + "-" + ingressPostfix
 	ingressRuleValue := ext_v1b1.IngressRuleValue{
 		HTTP: &ext_v1b1.HTTPIngressRuleValue{
 			Paths: []ext_v1b1.HTTPIngressPath{
```

Comment:
`ServiceName: string(serviceName),` below is incorrect. Name of the smith resource and name of the kube object, defined by that resource, are not the same thing. Can be, but don't have to be. This is a bug, can be fixed separately.

Reply:
I didn't change this, and I don't see it as part of this pull request. That said, I have clarified which variable is holding which value.